app/services/luhn_checksum.rb

```diff
@@ -0,0 +1,80 @@
+# Calculate the Luhn checksum of a number or a string that represents a number
+# Use the Luhn algorithm to calculate the checksum)
+#
+# Return true or false if a string (that represents a number) or number passes the Luhn checksum test
+#
+#
+# @url https://en.wikipedia.org/wiki/Luhn_algorithm
+# @url https://rosettacode.org/wiki/Luhn_test_of_credit_card_numbers
+#
+class LuhnChecksum
+
+  NO_DIGITS_IN_INPUT = 1
+
+  #
+  # Does the last digit of the number meet the Luhn checksum test?
+  #
+  # @param [ Integer | String ] integer_or_str to check
+  # @return Boolean - true if the checksum mod 10 == 0 (meets the test), else false
+  #   returns false for nil
+  #   returns false for an empty string
+  #   returns false if a string does not have any digits
+  #   removes any non-digit characters from the string (INCLUDING a PERIOD!) and then checks the resulting Integer
+  #
+  def self.valid?(integer_or_str)
```

Comment:
I don't see the advantage of developing our own class as the Orgnummer gem provides the functionality we need (via `new` and `valid?` methods).

This is a class that 1) has to check it's input (even though we are _not_ exposing this class to user input - it is only our code which calls this), and 2) provide it's own valid? algorithm which we get for "free" from the gem.

Then, we have a great deal of test code to maintain that is not needed if we leverage the gem.

If we had a use case for a general Luhn Checksum compliant numbers, then this would be great.  Since we don't, I just see it adding unneeded code to our code base.

Sorry to come across as harsh here, and I don't mean to be, but this is how I see it.

Reply:
I don't take that as "harsh" at all.  No worries.

I did this as part of understanding how orgnummer gem worked and added it in because it was just part of the whole thing _in my head._  But you're exactly right.  :-)